server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -160,6 +167,19 @@ public final class SnapshotsService extends AbstractLifecycleComponent implement
 
     private static final Logger logger = LogManager.getLogger(SnapshotsService.class);
 
+    /**
+     * We publish metrics of how many shards are in each of the following states
+     * these should be the list of statuses that potentially block movement in
+     * {@link SnapshotInProgressAllocationDecider}, or states that might delay
+     * a snapshot's completion.
+     */
+    private static final List<ShardState> TRACKED_SHARD_STATES = List.of(
```

Comment:
Is there a good reason for not just tracking all the states? I can see some value in knowing about completed shard snapshots too (e.g. to investigate delays in snapshot finalization)